tasmota/tasmota_xdrv_driver/xdrv_92_esp32_gnss.ino

```diff
@@ -0,0 +1,876 @@
+/*
+  xdrv_92_esp32_gnss.ino - GNSS with NMEA support and async web server
+  Copyright (C) 2025 Catalin Sanda
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  any later version.
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifdef ESP32
+#ifdef USE_GNSS
+
+/*********************************************************************/
+/*                          INCLUDES                                 */
+/*********************************************************************/
+#include <queue>
+#include <array>
+#include <cstring>
+#include <TinyGPSPlus.h>
+#include <GNSSParser.h>
+#include <vector>
+#include <algorithm>
+
+#ifdef USE_AUX_WEBSERVER
+#include <ESPAsyncWebServer.h>
+#endif
+
+#ifdef USE_NTRIP
+extern void ProcessRTCMMessage(const uint8_t *data, size_t length);
+#endif
+
+/*********************************************************************/
+/*                          DEFINES                                  */
+/*********************************************************************/
+#define XDRV_92 92
+#define D_CMND_GNSS "GNSS"
+#define RTC_UPDATE_INTERVAL (30 * 60) // Every 30 minutes
```

Comment:
In general, avoid general names for defines. Please use a unique prefix. Arduino merges all "*.ino" files into one huge C++ file, and collisions are likely